tree/dataframe/test/dataframe_snapshot_ntuple.cxx

```diff
@@ -0,0 +1,422 @@
+   const auto filename = "RDFSnapshotRNTuple_snap_templated.root";
+
+   auto df = ROOT::RDataFrame(fNtplName, fFileName);
+   auto sdf = df.Define("x", [] { return 10; }).Snapshot<float, int>("ntuple", filename, {"pt", "x"});
+
+   auto ntuple = RNTupleReader::Open("ntuple", filename);
+   EXPECT_EQ(1ull, ntuple->GetNEntries());
+
+   auto pt = ntuple->GetView<float>("pt");
+   auto x = ntuple->GetView<int>("x");
+
+   EXPECT_FLOAT_EQ(42.0, pt(0));
+   EXPECT_EQ(10, x(0));
+}
+
+TEST_F(RDFSnapshotRNTupleTest, DefaultToRNTupleJITted)
+{
+   const auto filename = "RDFSnapshotRNTuple_snap_jitted.root";
+
+   auto df = ROOT::RDataFrame(fNtplName, fFileName);
+   auto sdf = df.Define("x", [] { return 10; }).Snapshot("ntuple", filename, {"pt", "x"});
+
+   auto ntuple = RNTupleReader::Open("ntuple", filename);
+   EXPECT_EQ(1ull, ntuple->GetNEntries());
+
+   auto pt = ntuple->GetView<float>("pt");
+   auto x = ntuple->GetView<int>("x");
+
+   EXPECT_FLOAT_EQ(42.0, pt(0));
+   EXPECT_EQ(10, x(0));
+}
+
+TEST_F(RDFSnapshotRNTupleTest, ToTTreeTemplated)
+{
+   const auto filename = "RDFSnapshotRNTuple_to_ttree_templated.root";
+
+   auto df = ROOT::RDataFrame(fNtplName, fFileName);
+
+   RSnapshotOptions opts;
+   opts.fOutputFormat = ROOT::RDF::ESnapshotOutputFormat::kTTree;
+
+   auto sdf = df.Define("x", [] { return 10; }).Snapshot<float, int>("tree", filename, {"pt", "x"}, opts);
+
+   TFile file(filename);
+   auto tree = file.Get<TTree>("tree");
+   EXPECT_EQ(1ull, tree->GetEntries());
+
+   float pt;
+   int x;
+
+   tree->SetBranchAddress("pt", &pt);
+   tree->SetBranchAddress("x", &x);
+
+   tree->GetEntry(0);
+
+   EXPECT_FLOAT_EQ(42.0, pt);
+   EXPECT_EQ(10, x);
+}
+
+TEST_F(RDFSnapshotRNTupleTest, ToTTreeJITted)
+{
+   const auto filename = "RDFSnapshotRNTuple_to_ttree_jitted.root";
+
+   auto df = ROOT::RDataFrame(fNtplName, fFileName);
+
+   RSnapshotOptions opts;
+   opts.fOutputFormat = ROOT::RDF::ESnapshotOutputFormat::kTTree;
+
+   auto sdf = df.Define("x", [] { return 10; }).Snapshot("tree", filename, {"pt", "x"}, opts);
+
+   TFile file(filename);
+   auto tree = file.Get<TTree>("tree");
+   EXPECT_EQ(1ull, tree->GetEntries());
+
+   float pt;
+   int x;
+
+   tree->SetBranchAddress("pt", &pt);
+   tree->SetBranchAddress("x", &x);
+
+   tree->GetEntry(0);
+
+   EXPECT_FLOAT_EQ(42.0, pt);
+   EXPECT_EQ(10, x);
+}
+
+TEST_F(RDFSnapshotRNTupleTest, ScalarFields)
+{
+   auto df = ROOT::RDataFrame(fNtplName, fFileName);
+   auto sdf = df.Snapshot("ntuple", "RDFSnapshotRNTuple_scalar_fields.root", "pt");
+
+   std::vector<std::string> expected = {"pt"};
+   EXPECT_EQ(expected, sdf->GetColumnNames());
+}
+
+TEST_F(RDFSnapshotRNTupleTest, VectorFields)
+{
+   auto df = ROOT::RDataFrame(fNtplName, fFileName);
+   auto sdf = df.Snapshot("ntuple", "RDFSnapshotRNTuple_all_fields.root", "nnlo");
+
+   std::vector<std::string> expected = {"nnlo"};
+   EXPECT_EQ(expected, sdf->GetColumnNames());
+}
+
+TEST_F(RDFSnapshotRNTupleTest, ComplexFields)
```

Comment:
These tests with complex fields are very nice to have! Can we also include checks for the actual values in the output RNTuple?